enhancements/installer/aws-load-balancer-type.md

```diff
@@ -0,0 +1,318 @@
+
+## Summary
+
+The users would be able to specify the load balancer type either to elb or nlb during installation. 
+This will be used to deploy default ingress controller with user-specified load balancer service for OpenShift clusters
+running on AWS.
+Currently by default CLB/ELB is set by the cluster-ingress-operator.
+
+
+## Motivation
+
+Currently the default load balancer type is set to CLB/ELB. One can set default load balancer during installation using [this](https://docs.openshift.com/container-platform/4.10/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws-network-load-balancer.html#nw-aws-nlb-new-cluster_configuring-ingress-cluster-traffic-aws-network-load-balancer).
+However when the default IngressController CR is deleted, default ingress controller is recreated but as an CLB/ELB.
+This  behavior is expected as of now as the ingress operator by default creates CLB for default ingress controller. 
+Also if the CR is deleted then the operator wont be able check the information whether NLB is needed or not. 
+So, we need to store this information in the cluster object of ingress.
+
+### Goals
+
+As an administrator, I would like to specify the load balancer type in AWS either to NLB or CLB/ELB during installation.
+As an user/administrator, when an ingress controller CR is deleted then I would like the ingress controller to be recreated with the load balancer mentioned during installation.
+In short I would like the persistence of the load balancer type in the default ingress controller.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+We will be adding a field awsLBType in the install-config for the users to provide a load balancer either to NLB or CLB/ELB
+during installation.
+
+### Workflow Description
+
+The installer will accept a LB type from user and then we would set type of LB in the following object.
+
+`oc get ingress.config.openshift.io/cluster -o yaml`
+
+Then we would fetch info from this object and reconcile the operator to have the NLB changes reflected.
+
+### User Stories
+
+#### Story 1
+
+As an admin I want to install a specific load balancer rather than the default CLB/ELB.
+
+#### Story 2
+
+As an admin I want the load balancer type to persist even if the default ingress controller object is deleted which was mentioned during the installation.
+
+### API Extensions
+
+### Implementation Details/Notes/Constraints
+
+For the install config object we will be adding a field called awsLBType to the [API](https://github.com/openshift/installer/blob/master/pkg/types/aws/platform.go#L7)
+
+It will look as follows -
+```go
+package aws
+
+import "github.com/aws/aws-sdk-go/aws/endpoints"
+
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+
+// LBType allows user to set a load balancer type.
+// Allowed values are nlb or elb
+// +optional
+LBType string `json:"lbType,omitempty"`
+}
```

Comment:
Since this is an optional parameter, will clb/elb continue to be the default values?

Reply:
If nothing is set then the ingress operator will create a clb/elb.